### PR TITLE
add divise and aws cognito identity

### PR DESCRIPTION
**Pull request by likeuwill**
## Motivation and Context

Add ability to sign in via AWS Cognito without DataBase

## Checklist:

- [ ] It contains only changes required by issue (does not contain other PR)
- [ ] Includes link to a issue (if apply)
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class User
+  # required by Devise
```
Comment (miqs1992) [inline app/models/user.rb]:
```
current_user
=> super: no superclass method `serializable_hash' for #<User:0x00007f917e384098>
```

Comment (likeuwill) [reply app/models/user.rb]: @miqs1992 Where you get that error? In controller and view i have a current user.

File: config/routes.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
+  devise_for :users
```
[the remark this answers is not in the document]
Comment (likeuwill) [reply config/routes.rb]: rails routes

File: app/models/user.rb (hunk 1 above)
Comment (likeuwill) [reply app/models/user.rb]: current_user

File: config/initializers/remote_authenticable.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
```
[the remark this answers is not in the document]
Comment (likeuwill) [reply config/initializers/remote_authenticable.rb]: done

File: Gemfile [hunk 4]
```diff
@@ -7,37 +7,39 @@ ruby '2.6.3'
...
 
-gem 'puma', '~> 3.11'
-gem 'sass-rails', '~> 5'
-gem 'turbolinks', '~> 5'
-gem 'webpacker', '~> 4.0'
-
 gem 'activerecord-nulldb-adapter'
-gem 'aws-sdk-s3'
-gem 'bootsnap', '>= 1.4.2', require: false
+gem 'aws-sdk'
```
[the remark this answers is not in the document]
Comment (likeuwill) [reply Gemfile]: done

File: config/initializers/remote_authenticatable.rb [hunk 5]
```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+require 'aws-sdk'
```
[the remark this answers is not in the document]
Comment (likeuwill) [reply config/initializers/remote_authenticatable.rb]: removed

**Pull request merged**